apps/server/src/routes/routes.ts

```diff
@@ -224,6 +225,15 @@ function register(app: express.Application) {
     apiRoute(PST, "/api/password/change", passwordApiRoute.changePassword);
     apiRoute(PST, "/api/password/reset", passwordApiRoute.resetPassword);
 
+    // User management routes
+    apiRoute(GET, "/api/users/current", usersRoute.getCurrentUser);
+    apiRoute(GET, "/api/users/check-username", usersRoute.checkUsername);
+    route(GET, "/api/users", [auth.checkApiAuth, csrfMiddleware, auth.checkAdmin], usersRoute.getUsers, apiResultHandler);
+    route(GET, "/api/users/:userId", [auth.checkApiAuth, csrfMiddleware], usersRoute.getUser, apiResultHandler);
+    route(PST, "/api/users", [auth.checkApiAuth, csrfMiddleware, auth.checkAdmin], usersRoute.createUser, apiResultHandler);
+    route(PUT, "/api/users/:userId", [auth.checkApiAuth, csrfMiddleware], usersRoute.updateUser, apiResultHandler);
+    route(DEL, "/api/users/:userId", [auth.checkApiAuth, csrfMiddleware, auth.checkAdmin], usersRoute.deleteUser, apiResultHandler);
```

Comment:
These endpoints need to be used somewhere in the client. Preferably as a separate setting where the CRUD operations can take place.